app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -2,82 +2,88 @@
-                (SingleCellBaselineExperiment) experiment, contentType);
-        var otherExternalResourceLinks =  externalResourceType.equals("geo") ?
-                singleCellBaselineExperimentExternallyAvailableContentServiceGeo.list(
-                (SingleCellBaselineExperiment) experiment, contentType) :
-                externalResourceType.equals("ega") ?
-                        singleCellBaselineExperimentExternallyAvailableContentServiceEga.list(
-                        (SingleCellBaselineExperiment) experiment, contentType) :
-                        singleCellBaselineExperimentExternallyAvailableContentServiceEna.list(
-                        (SingleCellBaselineExperiment) experiment, contentType);
-        otherExternalResourceLinks.addAll(arrayExpressResourceLinks);
-        return otherExternalResourceLinks;
+
+        ImmutableList.Builder<ExternallyAvailableContent> arrayExpressResourceLinks = ImmutableList.builder();
+
+        if (experimentAccession.matches("E-MTAB.*|E-ERAD.*|E-GEUV.*")) {
+            arrayExpressResourceLinks.addAll(linkToArrayExpress.get(experiment));
+        }
+        var otherExternalResourceLinks =  externalResourceLinks(experiment);
+        arrayExpressResourceLinks.addAll(otherExternalResourceLinks.build());
+
+        return arrayExpressResourceLinks.build();
     }
 
-    private String getExternalResouceType(Experiment<?> experiment) {
-        var geoAccessions = experiment.getSecondaryAccessions().stream()
-                .filter(accession -> accession.matches("GSE.*"))
-                .collect(toImmutableList());
-        var egaAccessions = experiment.getSecondaryAccessions().stream()
-                .filter(accession -> accession.matches("EGA.*"))
-                .collect(toImmutableList());
+    private ImmutableList.Builder<ExternallyAvailableContent> externalResourceLinks(Experiment<?> experiment) {
+        ImmutableList.Builder<ExternallyAvailableContent> otherExternalResourceLinks = ImmutableList.builder();
+
+        Map<String, List<String>> resourceList = experiment.getSecondaryAccessions().stream()
+                .collect(Collectors.groupingBy(accession -> {
+                    if (accession.matches("GSE.*")) return "GEO";
+                    if (accession.matches("EGA.*")) return "EGA";
+                    if (accession.matches("PDX.*")) return "PRIDE";
+                    if (accession.matches("ERP.*|SRP.*|DRP.*|PRJEB.*|PRJNA.*|PRJDB.*")) return "ENA";
+                    return "OTHER";
+                }));
 
-        return geoAccessions.isEmpty() ? egaAccessions.isEmpty() ? "ena" : "ega" : "geo";
+        resourceList.replaceAll((k, v) -> v.stream().collect(toImmutableList()));
 
+        // Remove "OTHER" key if it exists
+        resourceList.remove("OTHER");
+
+        // Add to the builder based on the resource type
+        resourceList.forEach((resource, accessions) -> {
```

Comment:
Could you merge this iteration with the previous streaming operation into one, please?
That maybe even simplify the code in this method.
Thanks